frontend/src/main/scala/bloop/data/Project.scala

```diff
@@ -76,6 +85,30 @@ final case class Project(
     customWorkingDirectory.getOrElse(baseDirectory)
   }
 
+  /** Returns concatenated list of "sources" and expanded "sourcesGlobs". */
+  def allSourceFilesAndDirectories(): List[AbsolutePath] = {
+    val buf = mutable.ListBuffer.empty[AbsolutePath]
+    buf ++= sources
+    sourcesGlobs.foreach { glob =>
+      if (Files.isDirectory(glob.directory.underlying)) {
+        Files.walkFileTree(
```

Comment:
What would happen if this walk fails? e.g. another process could be going through the same tree and modifying it concurrently, which could cause this to throw exceptions. Would wrapping this in a `Task` make sense? I see the only call site is `BloopBspServices`.

Reply:
Good point. I updated this method to return a `Task`. I didn't add error handling but make the `Task` propagate to the return value in `BloopBspServices`.